copy/copy.go

```diff
@@ -1657,24 +1643,15 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 	}
 
 	// === Finally, send the layer stream to dest.
-	var uploadedInfo types.BlobInfo
-	// Note: the storage destination optimizes the committing of layers
-	// which requires passing the index of the layer.  Hence, we need to
-	// special case and cast.
-	dest, ok := c.dest.(internalTypes.ImageDestinationWithOptions)
-	if ok {
-		options := internalTypes.PutBlobOptions{
-			Cache:      c.blobInfoCache,
-			IsConfig:   isConfig,
-			EmptyLayer: emptyLayer,
-		}
-		if !isConfig {
-			options.LayerIndex = &layerIndex
-		}
-		uploadedInfo, err = dest.PutBlobWithOptions(ctx, &errorAnnotationReader{destStream}, inputInfo, options)
-	} else {
-		uploadedInfo, err = c.dest.PutBlob(ctx, &errorAnnotationReader{destStream}, inputInfo, c.blobInfoCache, isConfig)
+	options := internalTypes.PutBlobOptions{
```

Comment:
It _almost_ seems clear that `copyBlobFromStream` should have a `PutBlobOptions` parameter, rather than individual ones; the large number of parameters to `copyBlobFromStream` are not nice at all.

Except for `Cache`, and for the fact that we might, in that case, want to pass `PutBlobOptions` by reference instead of by value.

Oh well, we can always introduce a single-purpose struct just for naming parameters to `copyBlobFromStream`. But that’s out of scope.